packages/web-components/src/components/cta/__stories__/README.stories.mdx

```diff
@@ -0,0 +1,86 @@
+import { Preview, Props, Description, Story } from '@storybook/addon-docs/blocks';
+import contributing from '../../../../../../docs/contributing-license.md';
+import '../text-cta';
+
+# CTA
+
+> The CTA component is used to select from various CTA link types – text, cards,
+> and buttons – as well as defining their behavior, such as anchor links,
+> videos, and external links.
+
+<Preview withToolbar>
+  <Story id="components-cta--default" height="120px" />
+</Preview>
+
+## Getting started
+
+##### JS
+
+```javascript
+import '@carbon/ibmdotcom-web-components/es/components/cta/text-cta';
+```
+
+##### HTML
+
+```html
+<dds-text-cta href="https://www.ibm.com/" type="local">
+  Copy text
+</dds-text-cta>
+```
+
+## `<dds-text-cta>`
+
+`<dds-text-cta>` is the most basic version of CTA, presented as a simple link.
+
+<Props of="dds-text-cta" />
+
+## Working with a video player
+
+CTA components fire `dds-cta-run-action` custom event when user clicks on a CTA with `type="video"`.
+Appliction code can take care of creating/putting our video player component and working with our video player service API to put the video data into the video player component:
+
+```javascript
+import 'carbon-custom-elements/es/components/modal/modal-close-button';
+import VideoPlayerAPI from '@carbon/ibmdotcom-services/es/services/VideoPlayer/VideoPlayer';
+import '@carbon/ibmdotcom-web-components/es/components/cta/text-cta';
+import '@carbon/ibmdotcom-web-components/es/components/modal/modal';
+import '@carbon/ibmdotcom-web-components/es/components/lightbox-media-viewer/lightbox-media-viewer-body';
+import '@carbon/ibmdotcom-web-components/es/components/lightbox-media-viewer/lightbox-video-player';
+
+...
+
+document.addEventListener('dds-cta-run-action', async event => {
```

Comment:
This looks like a lot of lift for adopters to integrate a video player. I'm still not too far into the Typescript training, but wonder if there's something we could do to make it a bit easier for them to hook up a video?

Reply:
Good question @jeffchew - In story implementation there is `<dds-cta-container>` as an example implementation, but having our adopters refer to `<dds-cta-container>` implementation will make things `lit-element` specific. My thought is around adding Angular/React/Vue examples sometime later (like what `carbon-custom-elements` does) so we can provide examples tailored for the JavaScript framework of their choice (that our adopters can just copy/paste as the start).

Reply:
What I meant was what if we had something like:
```
<dds-text-cta videoId="abcd12345"></dds-text-cta>
```

This would then hook up the video for them as well as populate the text link with the video title?

Reply:
I'd like `<dds-text-cta>` focus on its UI. All-in-one text CTA can be done by `<dds-cta-container cta-style="text" .item={ copy: 'Copy text', href=''abcd12345', type='video' }>` (`.item` here means `item` property instead of `item` attribute)

Reply:
Given OOTB solution seems more desirable here, added an example with that approach: https://github.com/carbon-design-system/ibm-dotcom-library/pull/3128/commits/e71c373d17d4b7736e974316a422037e3ffd7312

Reply:
Thanks! I think this is probably what the majority of adopters would end up doing. Perhaps have the `<dds-cta-container>` approach first, then if you want to go into more details on having adopters customize the behavior, you can have something like `### Customized Actions Example` and then go into `dds-cta-run-action`?

Reply:
I see, updated the doc as such: https://github.com/carbon-design-system/ibm-dotcom-library/pull/3128/commits/90ecd0bb37871983b3177339117724182dafede4

Reply:
thank you! looks good!